### PR TITLE
Add __all__ attributes to icalendar modules

### LINKED ISSUE #638: [Python, Beginner] Add `__all__` attributes to modules.
The `__all__` variable is used to show which attributes of a module are public.
Please list all of the attributes that are imported in `icalendar/__init__.py` in their modules with the `__all__ = []` module variable.

See also:

- [`icalendar/__init__.py`](https://github.com/collective/icalendar/blob/main/src/icalendar/__init__.py)
- [What does `__all__` mean in Python?](https://stackoverflow.com/questions/44834/what-does-all-mean-in-python)

### PR DESCRIPTION
**Pull request by c0d33ngr**
This commit improves the namespace management within the `icalendar` directory by adding __all__ attributes to the `__init__.py` files of each module. This allows for clearer control over what is imported when using wildcard imports.
Closes #638

### CODE REVIEW

File: src/icalendar/__init__.py [hunk 1]
```diff
@@ -44,3 +44,9 @@
 
 # Switching the timezone provider
 from icalendar.timezone import use_pytz, use_zoneinfo
+
+
+__all__ = ['Calendar', 'Event', 'Todo', 'Journal', 'Timezone', 'TimezoneStandard', 'TimezoneDaylight', 'FreeBusy', 'Alarm',
+          'ComponentFactory', 'vBinary', 'vBoolean', 'vCalAddress', 'vDatetime', 'vDate', 'vDDDTypes', 'vDuration',
+          'vFloat, 'vInt', 'vPeriod','vWeekday', 'vFrequency', 'vRecur', 'vText', 'vTime', 'vUri', 'vGeo', 'vUTCOffset',
```
Comment (stevepiercy) [inline src/icalendar/__init__.py]: Typo fix.

This also needs a change log entry.

```suggestion
          'vFloat', 'vInt', 'vPeriod','vWeekday', 'vFrequency', 'vRecur', 'vText', 'vTime', 'vUri', 'vGeo', 'vUTCOffset',
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by stevepiercy** [comment]
See https://icalendar.readthedocs.io/en/latest/contributing.html#for-pull-requests-keep-this-in-mind

File: src/icalendar/__init__.py (hunk 1 above)
Comment (c0d33ngr) [reply src/icalendar/__init__.py]: I used github web so I couldn't spot the typo before pr. I've resolved it
sorry about that

**Comment by niccokunzmann** [comment]
Nice! Thank you!

**Pull request merged**